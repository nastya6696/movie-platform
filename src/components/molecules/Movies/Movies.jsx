import {ResultsFilter} from "../ResultsFilter";
import {SortingSelect} from "../SortingSelect";

import styles from './styles.module.scss';
import {MoviesList} from "../MoviesList";

export const Movies = ({movies}) => (
  <div className={styles.movies}>
    <div className={styles.content}>
      <div className={styles.sortingMethods}>
        <ResultsFilter/>
        <SortingSelect />
      </div>
      <p className={styles.resultText}><strong>{movies.length}</strong> movies found</p>
      <MoviesList movies={movies} />
    </div>
  </div>
)
