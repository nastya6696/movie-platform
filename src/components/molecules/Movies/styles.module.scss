@import "src/assets/scss/common";

.movies {
  background-color: $grey;
}

.content {
  margin-top: 10px;
  padding: 15px 60px;
  background-color: $black;
  color: $white;
}

.sortingMethods {
  display: flex;
  justify-content: space-between;
}

.resultText {
  @include fontSizes(20px, 25px);
  margin: 25px 0;
}
