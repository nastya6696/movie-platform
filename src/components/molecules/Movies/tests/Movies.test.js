import {Movies} from "../Movies";
import {create} from 'react-test-renderer';

jest.mock('../../ResultsFilter', () => ({
  ResultsFilter: (({ children }) => `
     (--- ResultsFilter ---): props {
     children: '${children}'
    }`
  )
}));

jest.mock('../../SortingSelect', () => ({
  SortingSelect: (() => `
     (--- SortingSelect ---)`
  )
}));

jest.mock('../../MoviesList', () => ({
  MoviesList: (({ movies }) => `
     (--- ResultsFilter ---): props {
     movies: '${movies}'
    }`
  )
}));

describe('Movies', () => {
  it('should create simple Movies component', () => {
    const movies = [
      {
        id: 1,
        title: 'movie1'
      },
      {
        id: 2,
        title: 'movie2'
      },
      {
        id: 3,
        title: 'movie3'
      }
    ]
    const component = create(<Movies movies={movies}/>);

    expect(component).toMatchSnapshot();
  })
});
