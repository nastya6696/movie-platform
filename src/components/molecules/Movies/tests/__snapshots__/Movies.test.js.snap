// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should create simple Movies component 1`] = `
<div
  className="movies"
>
  <div
    className="content"
  >
    <div
      className="sortingMethods"
    >
      
     (--- ResultsFilter ---): props {
     children: 'undefined'
    }
      
     (--- SortingSelect ---)
    </div>
    <p
      className="resultText"
    >
      <strong>
        3
      </strong>
       movies found
    </p>
    
     (--- ResultsFilter ---): props {
     movies: '[object Object],[object Object],[object Object]'
    }
  </div>
</div>
`;
